wfsim/strax_interface.py

```diff
@@ -160,6 +177,168 @@ def instruction_from_csv(filename):
+        else:
+            pass
+
+    def _instructions_from_epix_cut(self,):
+        '''Cut away events not within event_start & event_stop'''
+        mask=(self.instructions_epix['g4id']<self.context.config['event_stop'])&\
+             (self.instructions_epix['g4id']>=self.context.config['event_start'])
+        self.instructions_epix = self.instructions_epix[mask]
+
+
+    def instructions_from_epix(self,):
+        '''Run epix and save instructions as self.instruction.
+         For the moment we'll just process the whole file and then throw out all events we won't use
+         Epix needs to be imported in here to avoid circle imports'''
+        logging.info("Getting instructions from epix")
+        import epix
+
+        #make deepcopy cause some of the things in the setup will add unhashable stuff
+        epix_config = get_resource(self.context.config['epix_config'],fmt='json')
+        epix_config.update({'input_file':self.context.config['fax_file'],
+                            'entry_start':self.context.config['event_start'],
+                            'entry_stop':self.context.config['event_stop'],
+                            'source_rate':0})
+        epix_config = epix.run_epix.setup(epix_config)
+        self.instructions_epix=epix.run_epix.main(epix_config,return_wfsim_instructions=True)
+        self._set_max()
+        self._instructions_from_epix_cut()
+
+    def instructions_from_nveto(self,):
+        logging.info("Getting nveto instructions")
+        #TODO make config working with nveto...
+        self.context.config['nv_pmt_ce_factor'] = 1.0
+        self.instructions_nveto, self.nveto_channels, self.nveto_timings=read_optical(self.context.config)
+        
+
+    def set_timing(self,):
+        '''Set timing information in such a way to synchronize instructions for the TPC and nVeto'''
+        logging.info("Setting timings")
+        rate = self.context.config['event_rate']
+        start = self.context.config['event_start']
+        stop=self.context.config['event_stop']
+        n_events = stop-start
+
+        timings= (self.context.config['event_start']/rate + \
+                 wfsim.units.s*np.random.random(n_events)*(n_events)/rate).astype(np.int64)
+        timings.sort()
+
+        #For tpc we have multiple instructions per g4id. For nveto we only have one
+        counter=Counter(self.instructions_epix['g4id'])
+        i_per_id = [counter[k] for k in range(start,stop)]
+        timings_tpc=np.repeat(timings,i_per_id)
+        self.instructions_epix['time']+=timings_tpc
+        if self.context.config['neutron_veto']:
+            self.instructions_nveto['time']=timings[self.instructions_nveto['g4id']]
+    
+    def set_configuration(self,):
+        '''Set chunking configuration and feeds instructions to wfsim'''
+        max_events_per_chunk=500#is this hard coding really nessecairy...
+        chunk_size = np.ceil(max_events_per_chunk/self.context.config['event_rate'])
```

Comment:
Why do we need to know this apriori? I do not excaty know how the chunking works in WFSim. Is data which exceeds the current chunk postponed into the next one? Would not this otherwise lead to one of the following problems: 

* data overlapping into the next chunk
* data cut at chunk bounds

 I thought we would set the chunk start according to the end of the previous chunk.

Reply:
WFsim will break instructions up in those which are close together. These groups are simulated until it's over threshold. It wiill include the rest of the group